File: js/ts/h00.overriding/Rectangle.ts
///<reference path='Shape.ts' />
/******* RHINO/SPIDERMONKEY SUPPORT *******/
load('Shape.js');
/******* CLASS DECLARATION *******/
class Rectangle extends Shape {
  /******* CONSTRUCTORS *******/
  constructor(width : number, height : number) { super (width, height); }

  /******* METHODS *******/
  calculateArea() : number { return (this.width * this.height); }
}

File: objective-c/objc-2/i00.override/Rectangle.m
#import "Rectangle.h"

/**** CLASS IMPLEMENTATION *****/
@implementation Rectangle

/********* INITIALIZERS ********/
- (Rectangle *) initWithWidth: (NSNumber *) w height: (NSNumber *) h {
    if (self = [super init]) { width = w; height = h; }
    return self;
}

/*********** METHODS ***********/
- (NSNumber *) calculateArea { return [NSNumber numberWithFloat: ([width floatValue] * [height floatValue])]; }

@end
